src/main/groovy/com/github/jengelman/gradle/plugins/shadow/ShadowExtension.groovy

```diff
@@ -26,10 +26,11 @@ class ShadowExtension {
             publication.artifact(project.tasks.shadowJar)
         }
 
+        final def allDependencies = project.configurations.shadow.allDependencies
         publication.pom { MavenPom pom ->
             pom.withXml { xml ->
                 def dependenciesNode = xml.asNode().get('dependencies') ?: xml.asNode().appendNode('dependencies')
-                project.configurations.shadow.allDependencies.each {
+                allDependencies.each {
```

Comment:
Model objects such as `project`, `configurations`, etc, cannot be captured in the configuration cache. 

Instead, the code should capture just the relevant data (using collections or custom @immutable Groovy classes for example) in a `provider` (for lazy resolution):

```groovy
   final def allDependencies = provider { /* build a data structure with the relevant data here */ }
   ...
     pom.withXml { xml ->
       ...
       allDependencies.get().each { ...
 ```

Reply:
8b0bcd68f0d87fb052cafe34c014538bb98c3cf5, but still fails

Reply:
Also, see https://github.com/gradle/gradle/issues/24097#issuecomment-1452034244

Reply:
Thanks for both of you!

Reply:
Thanks to both of you!